class/practice4.1/src/main.cpp

```diff
@@ -0,0 +1,70 @@
+#include <builder.hpp>
+#include <map>
+#include <iostream>
+#include <string>
+#include <print>
+#include <fstream>
+#include <memory>
+std::map<std::string, double> readProducts(const std::string& filename){
+    std::map<std::string, double> products;
+    std::ifstream file(filename);
+    std::string line;
+    
+    while (getline(file, line)) {
+        std::size_t dashPos = line.find('-');
+        if (dashPos != std::string::npos) {
+            std::string name = line.substr(0, dashPos);
+            double price = std::stod(line.substr(dashPos + 1));
+            products[name] = price;
+        }
+    }
+
+    file.close();
+    return products;
+}
+
+std::string inputer(const std::string& message, const std::map<std::string, double>& availableProducts){
+    std::println("Pizza builder {}",message);
+    std::string product;
+    std::getline(std::cin, product);
+    auto it = availableProducts.find(product);
+    if (it != availableProducts.end()) {
+        return product;
+    }else{
+        return "nothing";
```

Comment:
maybe it is more reasonable to use std::optional<std::string> to report the case with empty input so it can be handled transparently on the higher level.